src/core/featuremodel.cpp

```diff
@@ -49,7 +49,7 @@ FeatureModel::ModelModes FeatureModel::modelMode() const
 
 void FeatureModel::setFeature( const QgsFeature &feature )
 {
-  if ( mModelMode != SingleFeatureModel || mFeature == feature )
+  if ( mModelMode != SingleFeatureModel )
```

Comment:
why did you remove this?

Reply:
To trigger a refresh when the feature is modified. If you look into FeatureModel::save(), you'll see it calls setFeature( modifiedFeature ), which ended up doing nothing as mFeature == modifiedFeature will be true even if the modifiedFeature has different attribute values et al.

Reply:
is it worth adding a `bool force = false` argument to the method?

Reply:
Na, I've seen other weird behaviors which is fixed by this removed check. So I'd prefer just removing it all together.

Reply:
>  as mFeature == modifiedFeature will be true even if the modifiedFeature has different attribute values et al.

According to `QgsFeatore::operator==` attributes should be compared as well. What did I miss?

Reply:
I don't think so no.

Reply:
https://github.com/qgis/QGIS/blob/master/src/core/qgsfeature.cpp#L68

I am no expert, but I would like to avoid having setters being executed too many times. It makes the debugging more difficult and might be noticeable by the user (rendering time, flickering).
but that's not my call to make here.

Reply:
I'll adjust.

Reply:
@3nids , done.